w_common/lib/src/common/managed_stream_subscription.dart

```diff
@@ -32,20 +32,20 @@ class ManagedStreamSubscription<T> implements StreamSubscription<T> {
   bool get isPaused => _subscription.isPaused;
 
   @override
-  Future<E> asFuture<E>([E futureValue]) {
+  Future<E> asFuture<E>([E? futureValue]) {
     return _subscription.asFuture(futureValue).whenComplete(_complete);
   }
 
   @override
-  Future<Null> cancel() {
+  Future<void> cancel() {
     var result = _subscription.cancel();
 
     // StreamSubscription.cancel() will return null if no cleanup was
     // necessary. This behavior is described in the docs as "for historical
     // reasons" so this may change in the future.
     if (result == null) {
```

Comment:
Result can no longer be null (`StreamSubscription.cancel` returns `Future<void>`), so we should be safe to remove this `if { }` block.